### ISSUE TITLE
[Intern Progress] Timi Joseph - Outreachy contribution Phase

### ISSUE THREAD

**Issue by Timi-Joseph** [ersilia-os/outreachy-contributions#30]
# 🌟 Intern Progress Tracker  
Welcome! Use this issue to track your progress as you work through the AI/ML modeling exercise. Follow each step, check the boxes when completed, and update your issue as you go.

# ▶️ **Entry tasks**
- [x] Motivation letter posted in this issue
- [x] Installed the Ersilia Model Hub and successfully run models!
- [x] Recorded system settigns in this issue
- [x] Explained completed tests of Ersilia in the issue

---

##  **1. Download a Dataset of Interest**
- [x] Created a fork of this repository to work on
- [x] Selected a dataset from Therapeutics Data Commons (TDC)
- [x] Verified that the dataset is a classification problem*
- [x] Read and understood the dataset’s background and endpoint  
- [x] Assessed computational feasibility (size, features, etc.)  
- [x] Downloaded the dataset using Python  
- [x] Saved data in the `/data` folder  
- [x] Started writing documentation in the README (Installation, dataset details, how to run code)  

---

##  **2. Featurise the Data**
- [x] Browsed Ersilia Model Hub for "Representation" models  
- [x] Selected an appropriate featuriser and justified my choice  
- [ ] Implemented featurisation in a notebook or script
- [ ] Ensured the process is reproducible 
- [ ] Updated the README with featurisation instructions  

---

##  **3. Build an ML Model**
- [ ] Selected an ML framework
- [ ] Trained the model on featurised data  
- [ ] Evaluated the model (accuracy, precision, recall, etc.)  
- [ ] Created graphs using Matplotlib for better visualization  
- [ ] Ensured all steps are reproducible  
- [ ] Documented the model-building process in README

---

##  **4. Prepare Code for Review**
- [ ] Double-checked that documentation is clear and complete  
- [ ] Evaluated if my model performance is satisfactory 
- [ ] Identified possible improvements  
- [ ] Updated this issue to request a review 

---

##  **5. Stretch Tasks (Optional)**
- [ ] Tried a different **featuriser**  
- [ ] Tested other **ML architectures**  
- [ ] Applied the model to **public datasets (e.g., ChEMBL)**  
- [ ] Any other improvements? (Describe below)  

---

## 📌 **Requesting Review**
Once you’ve completed **Steps 1-4**, request a review by mentioning a mentor:  
👤 *@GemmaTuron* Please review my work!  

---

### 📝 **Asking questions**
Use this issue to update your progress, explain where you are stuck or ask for review!

Thank you for contributing! 🎉

**Comment by GemmaTuron** [comment]
Hello @Timi-Joseph 

We are already on week 2 of the contribution period. If you aim to make an application to Ersilia I'd urge you to start on week 1 tasks as soon as possible.

**Comment by Timi-Joseph** [comment]
I am currently on it,

**Comment by Timi-Joseph** [comment]
# Motivation Letter

<b>Hello</b> @GemmaTuron 

I'm Timi Joseph, An Open Source Software Engineer, I studied and graduated as a Civil Engineer, but I have since been building a career in software engineering and development. Python is my go to tool for solving computational problems and building software solutions to business requirements.

I am excited to apply for this batch of the outreachy program as a means to get a platform to make real world contribution to problem, and also learn from and collaborate with other professionals in the field. Despite having a background in Engineering, i have often felt that i had plateaued at a stage of language foundation mastery without access to real world solving capabilities, I do hope that with the tasks and experience I would gain from the project, I would use that to better contribute to this project and the larger open Source community, Machine Learning and Large Data has been a skillset i adored alot, mostly because of the limitless possibilities it entail; Ersilia offers the perfect balance for python exploration skills and machine learning skills to my arsenal. I look forward to doing great things on this project.

Best regards,
Timi Joseph

**Comment by Timi-Joseph** [comment]
## Task  4: Obtain approval of the introductory tasks to continue contributing.

### Operating System Details

- **Operating System**: MacOS Sequoia
- **Installed RAM**: 16Gb
- **Storage**: 500GB
- **Chip**: Apple M1


### Ersilia and Docker Test Description

#### Tasks
- [x] installation of Ersilia Locally
- [x] Run Selected Models on a specifed Molecule

#### Steps
- Installed Python: Already had Python installed Locally
  ```bash
  python3 --version
  # output: Python 3.12.9
  ```
- Installed Anaconda: Downloaded and Install Anaconda for Mac from [Here](https://www.anaconda.com/docs/main)
   ```bash
   conda --version
   # output: conda 24.9.2
   ```
- Installed Git CLI: Already had Git CLI installed Locally
   ```bash
   git --version
  # output: git version 2.48.1
   ```
- Installed Git-lft
  ```bash
   git-lfs --version
  # output: git-lfs/3.6.1 (GitHub; darwin arm64; go 1.23.4)
  ```
- Installed Docker from [here](https://www.docker.com/)
  ```bash
  docker --version
  # output:  Docker version 28.0.1, build 068a01e
  ```
- Install Ersilia
   ```bash
   ersilia --version
   # output: ersilia, version 0.1.42
   ```

#### Running Tests
The General flow of running a model (test) is as follows;
```bash
ersilia -v fetch <model_id>
ersilia serve model_id
ersilia -v run -i <molecule>
```

1. Run Test on Model eos3b5e
   ```bash
   # Molecule used: "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O"
   # output:
   [
          {
              "input": {
                  "key": "HEFNNWSXXWATRW-UHFFFAOYSA-N",
                  "input": "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O",
                  "text": "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O"
              },
              "output": {
                  "outcome": [
                      206.28499999999997
                  ]
              }
          }
      ]
   ```

2. Run Test on Model eos2r5a
   - Issue fetching the model: Stack trace of issue posted [here](https://gist.github.com/Timi-Joseph/b52282c9042eeb0e74c99c10a9a5e09f) [Resolved ✅ [^1]]
     ```bash
         {
          "input": {
              "key": "HEFNNWSXXWATRW-UHFFFAOYSA-N",
              "input": "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O",
              "text": "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O"
          },
          "output": {
              "outcome": [
                  0.9899624
              ]
          }
      }
     ```

3. Run Test on Model eos3b5e
   ```bash
   # Molecule used: "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O"
   # output:
   [
        {
            "input": {
                "key": "HEFNNWSXXWATRW-UHFFFAOYSA-N",
                "input": "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O",
                "text": "CC(C)CC1=CC=C(C=C1)C(C)C(=O)O"
            },
            "output": {
                "outcome": [
                    2.191755
                ]
            }
        }
    ]
   ```


[^1]: Due to a possible issue related to the image on Mac Silicon Chip, i had to run the model from inside docker desktop dashboard.

**Comment by GemmaTuron** [comment]
Hi @Timi-Joseph !

You have marked the data selection and so but I did not see any summary here! Please make sure to update the issue so I can provide feedback

**Comment by Timi-Joseph** [comment]
# Outreachy contribution period

## Installation
Tools needed;

- Python3 Installation on the host machine
- Vscode enabled with Jupyter support or Google Colab or Jupyter notebook
- Git
- Create And Activate Python Virtual Environment
  ```bash
  python3 -m venv venv && source venv/bin/activate && pip install requirements.txt
  ```
- Install libmp: Needed for Macos to run XGBoost package
  ```bash
  brew install libomp
  ```

## Dataset of Interest Details
Drug-induced liver injury (DILI) is fatal liver disease caused by drugs and it has been the single most frequent cause of safety-related drug marketing withdrawals for the past 50 years (e.g. iproniazid, ticrynafen, benoxaprofen). This dataset is aggregated from U.S. FDA’s National Center for Toxicological Research.

### Dataset Analysis:
This dataset is a binary classification task, where the goal is to predict the activity of BBB against a drug. It is positive when Y=1 and negative when Y=0.

#### Features:

Drug_ID: Assigns a reference ID to each drug

Drug : SMILES representation.

(Y): predict whether it can cause liver injury (1) or not (0), explicitly provided for each compound.

#### Class Distribution:
Total Samples: 475
Y=1: 236 drugs (49.7% of the dataset).
Y=0: 239 drugs (50.3% of the dataset).

Dataset Checks:
1. [X] Dataset is a Classification Task: Output (Y) is 1 or 0
3. [X] Dataset is computationally feasible and is made up of 475 Compounds

### Download and Save Dataset
Download the Dataset from [Therapeutics Data Commons](https://tdcommons.ai/single_pred_tasks/tox#ames-mutagenicity). The DILI (Drug Induced Liver Injury) Dataset

- Open the [drug_induced_liver_injury.ipynb](notebooks/drug_induced_liver_injury.ipynb) Notebook and run each cell the to download the dataset

    1. The notebook install TDC library using
        ```bash
        !pip install PyTDC
        ```
    2. Load the AMES Dataset with the library
       ```python
       from tdc.single_pred import Tox
       data = Tox(name = 'Dili')
       ```
    3. Split the dataset into Train, Validation and Test sets
        ```python
        split = data.get_split()
        ```
    This gave me 5094 Train data, 1456 Validation data and 738 test data

    4. Save the Downloaded dataset to the project environment data directory
       ```python
        split['train'].to_csv("../data/dili/dili_train.csv", index=False)
        split['valid'].to_csv("../data/dili/dili_valid.csv", index=False)
        split['test'].to_csv("../data/dili/dili_test.csv", index=False)
       ```
    This saves the downloaded dataset in an Ames folder under the data directory at the root of the project.

**Comment by GemmaTuron** [comment]
Hi @Timi-Joseph 

As I mentioned several times in the Slack channel I'll only be able to provide feedback on w2 and w3 tasks until last Monday, to focus only in final applications during the week. If you still want to make a final application to Ersilia please complete the tasks as described and start preparing the Outreachy final application. Thanks!

**Comment by miquelduranfrigola** [comment]
Auto-closing this Outreachy intern-progress tracker as completed — the contribution round has ended. Reopen if follow-up is needed.
